### PR TITLE
use some string interpolation

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs [hunk 1]
```diff
@@ -426,7 +423,7 @@ public IEnumerable<TestMethod> BuildFrom(IMethodInfo method, Test? suite)
                 if (_untilDate > DateTimeOffset.UtcNow)
                 {
                     test.RunState = RunState.Ignored;
-                    string reason = string.Format("Ignoring until {0}. {1}", _untilDate.Value.ToString("u"), IgnoreReason);
+                    string reason = $"Ignoring until {_untilDate.Value.ToString("u")}. {IgnoreReason}";
```
Comment (jnm2) [inline src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs:426]: I helped with a VS light bulb suggestion that should be offering to change to:
```suggestion
                    string reason = $"Ignoring until {_untilDate.Value:u}. {IgnoreReason}";
```

File: src/NUnitFramework/framework/Interfaces/TNode.cs [hunk 2]
```diff
@@ -347,7 +347,7 @@ private static NodeList ApplySelection(NodeList nodeList, string xpath)
 
         private static string CharToUnicodeSequence(char symbol)
         {
-            return string.Format("\\u{0}", ((int)symbol).ToString("x4"));
+            return $"\\u{((int)symbol).ToString("x4")}";
```
Comment (jnm2) [inline src/NUnitFramework/framework/Interfaces/TNode.cs:350]:
```suggestion
            return $"\\u{(int)symbol:x4}";
```

File: src/NUnitFramework/framework/Internal/Builders/PairwiseStrategy.cs [hunk 3]
```diff
@@ -573,7 +573,7 @@ private void SelfTest( List<TestCaseInfo> testCases )
 
                                 if ( !IsTupleCovered( testCases, tuple ) )
                                 {
-                                    throw new InvalidOperationException( string.Format( "PairwiseStrategy : Not all pairs are covered : {0}", tuple.ToString() ) );
+                                    throw new InvalidOperationException($"PairwiseStrategy : Not all pairs are covered : {tuple.ToString()}");
```
Comment (jnm2) [inline src/NUnitFramework/framework/Internal/Builders/PairwiseStrategy.cs:576]:
```suggestion
                                    throw new InvalidOperationException($"PairwiseStrategy : Not all pairs are covered : {tuple}");
```

File: src/NUnitFramework/framework/Internal/Extensions/IPropertyBagDataExtensions.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@ internal static class IPropertyBagDataExtensions
         /// <param name="reason">The reason the test is being ignored until that date</param>
         internal static void AddIgnoreUntilReason(this IPropertyBag properties, DateTimeOffset untilDate, string reason)
         {
-            string skipReason = string.Format("Ignoring until {0}. {1}", untilDate.ToString("u"), reason);
+            string skipReason = $"Ignoring until {untilDate.ToString("u")}. {reason}";
```
Comment (jnm2) [inline src/NUnitFramework/framework/Internal/Extensions/IPropertyBagDataExtensions.cs:23]:
```suggestion
            string skipReason = $"Ignoring until {untilDate:u}. {reason}";
```

**Review by jnm2 — APPROVED** [review]

**Comment by SimonCropp** [comment]
can we get a merge here?

**Pull request merged**